### PR TITLE
Use int32_t instead of int for train_index

### PR DESCRIPTION
**Pull request by tupaschoal**
As proposed by @duncanspumpkin on #10135

> int32_t (not sure why trainIndex has been defined as an int should have been int32_t)

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -2215,7 +2215,7 @@ static void train_ready_to_depart(rct_vehicle* vehicle, uint8_t num_peeps_on_tra
     vehicle->SetState(VEHICLE_STATUS_WAITING_FOR_PASSENGERS);
 }
 
-static int ride_get_train_index_from_vehicle(Ride* ride, uint16_t spriteIndex)
+static int32_t ride_get_train_index_from_vehicle(Ride* ride, uint16_t spriteIndex)
 {
     uint32_t trainIndex = 0;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp]: change this type to `int32_t` as well

Comment (tupaschoal) [reply src/openrct2/ride/Vehicle.cpp]: I had, actually, but when comparing it to `std::size(ride->num_vehicles)` it errored out due to narrowing. I thought about doing `static_cast` but was unsure. What do you think?

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp]: What about refactoring the whole thing to use a std::optional? It would then get rid of the super edge cases.

Comment (tupaschoal) [reply src/openrct2/ride/Vehicle.cpp]: Will try this tonight, if it's too big

Comment (tupaschoal) [reply src/openrct2/ride/Vehicle.cpp]: I might have misunderstood, did you mean change the `ride->vehicles` to be an optional?
I think it only errored on the `else if (trainIndex >= std::size(ride->vehicles))`, not sure about the `if (trainIndex >= ride->num_vehicles)`. Sorry for the confusion.

Comment (duncanspumpkin) [reply src/openrct2/ride/Vehicle.cpp]: I was meaning for the return of the function. It gets rid of the invalid train index.

**Comment by duncanspumpkin** [comment]
I've updated this to use std::optional for the return.

**Pull request merged**